shims/common/src/main/scala/org/apache/gluten/config/GlutenConfig.scala

```diff
@@ -685,6 +685,13 @@ object GlutenConfig {
     .stringConf
     .createWithDefault("")
 
+  val GLUTEN_SUPPORTED_COLLAPSED_FUNCTIONS =
+    buildConf("spark.gluten.sql.supported.collapseNestedFunctions")
+      .internal()
+      .doc("Collapse nested functions as one for optimization.")
+      .stringConf
+      .createWithDefault("and,or");
```

Comment:
@KevinyhZou, do you find any unsuitable corner cases? E.g., wrong result, performance degradation. If no, can we always enable this optimization without introducing a config?

Reply:
Not found unsuitable case for the ci testing. But I'm afraid there maybe some unsuitable case in our online sqls, which the ci testing do not cover，So I think it‘s better to keep them